platform/core.multitabs/src/org/netbeans/core/multitabs/impl/FolderNameTabDecorator.java

```diff
@@ -54,10 +79,14 @@ public String getText( TabData tab ) {
                 if( fo.isData() ) {
                     FileObject folder = fo.getParent();
                     if( null != folder ) {
-                        String folderName = folder.getNameExt() + pathSeparator;
-                        String defaultText = tab.getText();
-
-                        return merge( folderName, defaultText );
+                        String folderName = folder.getNameExt() + File.separator;
+                        // don't use faded colors in tabs when colored tabs are active
+                        // since it is difficult to get right with so many colors and shades involved
+                        // the switcher uses a line as marker instead of bg change and can continue using fade colors
+                        if (!(isTab && settings.isSameProjectSameColor())) {
+                            folderName = "<font color=\"" + fadeColor + "\">" + folderName + "</font>"; //NOI18N
+                        }
+                        return merge(folderName, tab.getText());
```

Comment:
the path decoration will now disable itself on tabs if colored tabs are active, the switcher will keep using the faded color since its colored tabs impl is different

an attempt to fix some contrast problems is pending at https://github.com/apache/netbeans/pull/8175

Reply:
project tab colors off:
![image](https://github.com/user-attachments/assets/23735e68-4e44-46ce-af7b-d73f96c92f13)

project tab colors on:
![image](https://github.com/user-attachments/assets/ff4e8a04-90dd-4700-a19c-7a739aba33fa)